python/ray/tune/execution/ray_trial_executor.py

```diff
@@ -606,20 +606,22 @@ def stop_trial(
             out = self._find_future(trial)
             for result_id in out:
                 self._futures.pop(result_id)
+        trial.saving_to = None
 
     def continue_training(self, trial: Trial) -> None:
         """Continues the training of this trial."""
         self._train(trial)
 
-    def pause_trial(self, trial: Trial) -> None:
+    def pause_trial(self, trial: Trial, should_checkpoint: bool = True) -> None:
```

Comment:
Should I avoid having this be configurable? I think a lot of other places in the code depend on a paused trial always having an in-memory checkpoint. This one case of PBT will never use the trial's own in-memory checkpoint, which is why I would want this as False.

Reply:
I think this is fine - it's the main reason why we call `stop` and then `set_status` in PBT at the moment, and I think this is cleaner.

Reply:
Could we add some documentation here?